static/js/narrow.js

```diff
@@ -771,7 +771,7 @@ export function maybe_add_local_messages(opts) {
 }
 
 export function update_selection(opts) {
-    if (message_list.narrowed.empty()) {
+    if (message_lists.current.empty()) {
         return;
```

Comment:
I think this change is incorrect, but maybe the code was bugged regardless. 

This function gets called from a continuation, and so in theory `message_lists.current` can have changed due to unnarrow in the meantime. I would propose that we do a preparatory refactor to pass `msg_list` to `update_selection`, and if that is no longer the current message list, return early.

If we do that, then I think this code path is safe.

Reply:
Okay, done!